### PR TITLE
Enable CPI extensions

### PR DESCRIPTION
**Pull request by EleanorRigby**
[#157263518](https://www.pivotaltracker.com/story/show/157263518)

### CODE REVIEW

**Comment by EleanorRigby** [comment]
@nehagjain15 : do you think it will be good idea to move all extension manager stuff in a separate module? I am not liking the way I am putting all things in cloud.rb. Thoughts?

File: src/vsphere_cpi/lib/cloud/vsphere/cloud.rb [hunk 1]
```diff
@@ -41,6 +44,44 @@ def enable_telemetry
       other_client.logout rescue nil
     end
 
+    def enable_extension
+      http_client = VSphereCloud::CpiHttpClient.new
```
Comment (nehagjain15) [inline src/vsphere_cpi/lib/cloud/vsphere/cloud.rb]: do you need a separate client for this?

**Comment by nehagjain15** [comment]
That sounds great to move it out from cloud.rb

File: src/vsphere_cpi/lib/cloud/vsphere/cloud.rb [hunk 2]
```diff
@@ -41,6 +44,44 @@ def enable_telemetry
       other_client.logout rescue nil
     end
 
+    def enable_extension
```
Comment (nehagjain15) [inline src/vsphere_cpi/lib/cloud/vsphere/cloud.rb]: Should this method be named create_cpi_extension instead as you are either returning and existing/creating a new extension right?

File: src/vsphere_cpi/lib/cloud/vsphere/cloud.rb (hunk 1 above)
Comment (EleanorRigby) [reply src/vsphere_cpi/lib/cloud/vsphere/cloud.rb]: Just for the fact that it is a separate operation like adding telemetry configuration and keeping it asynchronous helps to speed up things. It takes sometime to register and find an extension.

File: src/vsphere_cpi/lib/cloud/vsphere/cloud.rb (hunk 2 above)
Comment (EleanorRigby) [reply src/vsphere_cpi/lib/cloud/vsphere/cloud.rb]: yep, I can rename that.

File: src/vsphere_cpi/lib/cloud/vsphere/cpi_extension.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    def register_extension(extension)
+      ext_mgr = vc_client.service_content.extension_manager
+      ext_mgr.register_extension(extension)
+    end
+
+    def extension_is_registered?(key: DEFAULT_VSPHERE_CPI_EXTENSION_KEY)
+      ext_mgr = vc_client.service_content.extension_manager
+      cpi_extension = ext_mgr.find_extension(key)
+      logger.debug("Extension already exists")
```
Comment (nehagjain15) [inline src/vsphere_cpi/lib/cloud/vsphere/cpi_extension.rb]: shouldn't this log only if !cpi_extencion.nil? else it would be misleading

**Pull request merged**